src/cmakeProject.ts

```diff
@@ -1650,8 +1650,8 @@ export class CMakeProject {
                         } else {
                             buildLogger.info(localize('build.finished.with.code', 'Build finished with exit code {0}', rc));
                         }
-                        const fileDiags = consumer!.compileConsumer.resolveDiagnostics(drv!.binaryDir);
-                        if (fileDiags) {
+                        const fileDiags: FileDiagnostic[] | undefined = drv!.config.parseBuildDiagnostics ? consumer!.compileConsumer.resolveDiagnostics(drv!.binaryDir) : [];
+                        if (fileDiags && fileDiags.length >= 1) {
```

Comment:
If you require the length to be greater than one, doesn't that block us from clearing the problems?  My point of setting it to `[]` was so that `populateCollection` would be called.